cmd/backup/script.go

```diff
@@ -503,7 +504,12 @@ func (s *script) createArchive() error {
 		return fmt.Errorf("createArchive: error walking filesystem tree: %w", err)
 	}
 
-	if err := createArchive(filesEligibleForBackup, backupSources, tarFile, s.c.BackupCompression.String()); err != nil {
+	concurrency := s.c.GzipParallelism.Int()
+	if concurrency == 0 {
+		concurrency = runtime.GOMAXPROCS(0)
```

Comment:
Can this have possible side effects on following operations? Also, should it only apply when `gz` is the target compression format? Right now, this would also apply when using `zstd` compression.

I.e. should this happen further down the call stack in `compress` itself?

Reply:
Preferably, I'd like to put the `switch` case in `archive.go` for selecting and initializing the respective compression backend in its own function to unclutter the compress function there. It should only care about compression, not what backend is used there, as everything after the initialization is the same. Just to use the opportunity to clean up in there.